crates/api/src/site/mod_log.rs

```diff
@@ -42,69 +53,269 @@ impl Perform for GetModlog {
 
     check_private_instance(&local_user_view, context.pool()).await?;
 
+    let type_ = data.type_.unwrap_or(ModlogActionType::All);
     let community_id = data.community_id;
-    let mod_person_id = data.mod_person_id;
+
+    let site = blocking(context.pool(), Site::read_local_site).await??;
+    let is_admin = is_admin(local_user_view.as_ref().expect("")).is_ok();
+    let local_person_id = match local_user_view {
+      Some(s) => s.person.id,
+      None => PersonId(-1),
```

Comment:
Why are you setting these to -1? They should be left as options so you can catch the None case. Also this is a better syntax:

`let local_person_id = local_user_view.map(|l| l.person.id);`

Reply:
It is only used to clean up the syntax around calling 'is_mod_or_admin' which requires a PersonId.